pkg/base/common/extension/registry.go

```diff
@@ -0,0 +1,21 @@
+package extension
+
+import "github.com/transaction-wg/seata-golang/pkg/base/registry"
+
+var (
+	registrys = make(map[string]func() (registry.Registry, error))
+)
+
+// SetRegistry sets the registry extension with @name
+func SetRegistry(name string, v func() (registry.Registry, error)) {
+	registrys[name] = v
+}
+
+// GetRegistry finds the registry extension with @name
+func GetRegistry(name string) (registry.Registry, error) {
+	if registrys[name] == nil {
+		panic("registry for " + name + " does not exist.")
```

Comment:
since this func has provided error as one of the results, maybe we can return this error instead of panic?